src/slurm_plugin/clustermgtd.py

```diff
@@ -1126,36 +885,123 @@ def _handle_protected_mode_process(self, slurm_nodes, partitions, bootstrap_fail
         # Partition does not have job running
         log.info(
             "Partitions bootstrap failure count: %s, cluster will be set into protected mode if "
-            "protected failure count reach threshold",
+            "protected failure count reach threshold.",
             self._partitions_protected_failure_count_map,
         )
-        partitions_above_threshold = [
-            part
-            for part, failures in self._partitions_protected_failure_count_map.items()
-            if part not in self._inactive_partitions and failures >= self._config.protected_failure_count
-        ]
-        if partitions_above_threshold:
-            log.info("Bootstrap failure partitions: %s", print_with_count(partitions_above_threshold))
-            nodename_to_slurm_nodes_map = {node.name: node for node in slurm_nodes}
-            partitions_to_disable = ClusterManager._get_partitions_to_disable(
-                nodename_to_slurm_nodes_map, partitions_above_threshold, partitions
+
+        partitions_to_disable = []
+        bootstrap_failure_partitions_have_jobs = []
+        for part, failures in self._partitions_protected_failure_count_map.items():
+            if part.is_active and failures >= self._config.protected_failure_count:
+                if part.has_running_job():
+                    bootstrap_failure_partitions_have_jobs.append(part)
+                else:
+                    partitions_to_disable.append(part)
+
+        if partitions_to_disable:
+            log.info("Bootstrap failure partitions: %s", print_with_count(partitions_to_disable))
+        if bootstrap_failure_partitions_have_jobs:
+            log.info(
+                "Bootstrap failure partitions %s currently have jobs running, not disabling them.",
+                print_with_count(bootstrap_failure_partitions_have_jobs),
             )
+        if partitions_to_disable or bootstrap_failure_partitions_have_jobs:
```

Comment:
Why do we need `bootstrap_failure_partitions_have_jobs` here?

Reply:
Because in the function enter_protected_mode, if there's no`partitions_to_disable`, log.info("Not entering protected mode since active are jobs running in bootstrap failure partitions")
It consider various of cases:
* case1: Bootstrap failure partitions: [A, B, C, D]We may have [C,D] partitions to disable, [A,B] partition have job running but bootstrap failure
* case2: [A,B, C, D]partition are all the bootstrap failure partitions, but they all have jobs running